src/main/java/Duke.java

```diff
@@ -1,10 +1,271 @@
+                }
+                break;
+            case COMMAND_TODO:
+                try {
+                    if (line.equals("todo")) {
+                        throw new EmptyDescriptionException(line);
+                    } else {
+                        String description = line.substring(line.indexOf("todo") + 5);
+                        if (description.length() < 1) {
+                            throw new EmptyDescriptionException(description);
+                        } else {
+                            Todo td = new Todo(description);
+                            tasksList.add(td);
+                            System.out.println("Got it. I've added this task: " + System.lineSeparator() +
+                                    td + System.lineSeparator() + "Now you have " + tasksList.size() + " tasks in the list");
+                            String textToAdd = "T | " + td.status + " | " + td.description + System.lineSeparator();
+                            taskOverrideList.add(textToAdd);
+                            appendToFile(textFilePath, textToAdd);
+                        }
+                    }
+                } catch (EmptyDescriptionException e) {
+                    System.out.println(e.getExceptionMessage());
+                } catch (IOException f) {
+                    System.out.println("Something went wrong. I am unable to add the todo task.");
+                }
+                break;
+            case COMMAND_DEADLINE:
+                try {
+                    if (line.equals("deadline")) {
+                        throw new EmptyDescriptionException(line);
+                    } else {
+                        String message = line.substring(line.indexOf("deadline") + 9);
+                        if (message.length() < 1) {
+                            throw new EmptyDescriptionException(message);
+                        } else if (message.indexOf('/') == -1 || !message.contains("by")) {
+                            throw new UnrecognisedDeadlineException(message);
+                        } else {
+                            String description = line.substring(line.indexOf("deadline") + 9, line.indexOf("/"));
+                            String by = line.substring(line.indexOf("by") + 3);
+                            Deadline d = new Deadline(description, by);
+                            tasksList.add(d);
+                            System.out.println("Got it. I've added this task: " + System.lineSeparator() +
+                                    d + System.lineSeparator() + "Now you have " + tasksList.size() + " tasks in the list");
+                            String textToAdd = "D | " + d.status + " | " + d.description + "| " + by + System.lineSeparator();
+                            taskOverrideList.add(textToAdd);
+                            appendToFile(textFilePath, textToAdd);
+                        }
+                    }
```

Comment:
Likewise, you could consider whether to abstract out this section of code to avoid deep nesting of code.